src/frontend/js/components/Modal/index.spec.tsx

```diff
@@ -50,4 +55,50 @@ describe('<Modal />', () => {
     expect($modal).toBeInstanceOf(HTMLDivElement);
     expect($overlay).toBeInstanceOf(HTMLDivElement);
   });
+
+  it('displays close button', () => {
+    render(
+      <IntlProvider locale="en">
+        <Modal isOpen={true} />
+      </IntlProvider>,
+    );
+
+    const $closeButton = document.querySelector('.modal .modal__closeButton');
+    expect($closeButton).toBeInstanceOf(HTMLElement);
+  });
+
+  it('has no close button', () => {
+    render(
+      <IntlProvider locale="en">
+        <Modal isOpen={true} hasCloseButton={false} />
+      </IntlProvider>,
+    );
+
+    const $closeButton = document.querySelector('.modal .modal__closeButton');
+    expect($closeButton).toBeNull();
+  });
+
+  it('displays title', () => {
+    render(
+      <IntlProvider locale="en">
+        <Modal isOpen={true} title={<div className="custom__header">Hi there</div>} />
+      </IntlProvider>,
+    );
+
+    const $customHeader = document.querySelector('.modal .modal__header .custom__header');
```

Comment:
ditto

tips : You should put an attribute `data-testId` to your custom title element then retrieve the element through `getByTestId`.